Support EdDSA

Tougee opened the pull request.

Ephenodrom committed:
Add ASN1Parser

Ephenodrom committed:
Fix Unit tests for ASN1 Sequence and Set

Commit:
Update chacha20.dart

Commit:
Update chacha7539.dart

Ephenodrom committed:
Add constructed behavior

Tougee committed:
Basic EdDSA

AKushWarrior commented:
@Tougee The implementation is good, but the tests need some work. When you sign and verify, you don't necessarily prove that the output is correct, just that signing and verification are symmetrically incorrect/correct.

Ideally, the test would also check that the signature is aligned with the EdDSA standard's output.

Perhaps you could run the test case through another language's EdDSA implementation, or look at the definition paper (where EdDSA's algorithm is defined) for test cases.

Ephenodrom committed:
improve BER encoding/decoding

mwcw committed:
Updated pana invocation.

mwcw committed:
Merge branch 'chacha-external' of https://github.com/CDDelta/pc-dart into CDDelta-chacha-external

mwcw committed:
Removed empty line at end of file.

mwcw committed:
Merge branch 'CDDelta-chacha-external' into master

Ephenodrom committed:
add der ber encoding

Tougee committed:
Export EdDSASigner

Ephenodrom committed:
Rename and export of the asn1 library

Ephenodrom committed:
revert renaming

Ephenodrom committed:
improve object identifier

Tougee committed:
Update ed25519_edwards version

AKushWarrior commented:
Also, we can't merge commits that have external dependencies. So you'll have to internalize that thing you added to pubspec.yaml.

Tougee commented:
Yeah, got it. 
It's just an experimental branch for now, I am working on lots of stuffs related to a [jose](https://github.com/Tougee/jose/pull/1) library based on pc-dart.
I'll refine tests and other things if it is a PR for original pc-dart project.

Tougee committed:
Add tests from RFC 8032

mwcw committed:
Merge branch 'master' of https://github.com/Ephenodrom/pc-dart into Ephenodrom-master

mwcw committed:
Thanks!

Merge branch 'Ephenodrom-master' into master

mwcw committed:
#45

Added well named getters for public and private exponents and associated tests.
Added public key exponent field in private key and associated tests.
Updated equality methods, added associated tests.
Updated change log.

Ephenodrom committed:
Fix linters

Ephenodrom committed:
Add ASN1TeltetextString

Ephenodrom committed:
Add add() method to the ASN1Sequence

Ephenodrom committed:
Handle unused bits

Ephenodrom committed:
Add OID

Ephenodrom committed:
Add ASN1GeneralizedTime

Ephenodrom committed:
Add asn1 dump feature

Ephenodrom committed:
Fix encoding of ASN1Integer with BigInt.from(0)

Ephenodrom committed:
minor improvements

mwcw committed:
Merge branch 'Ephenodrom-master' into master

mwcw committed:
Updated changelog for 2.0.0-rc2

mwcw committed:
Pubspec 2.0.0-rc2

mwcw committed:
Updated to 2.0.0 -- updated and fixed wrong date in change log.

alextekartik committed:
fix: remove dart:io dependency to allow compiling for web/flutter_web

hoylen committed:
RSAPrivateKey calculates the public exponent from the other values.

hoylen committed:
Added check for RSA modulus matches p and q.

hoylen committed:
Fixed RSA key generator tests to use valid keys.

Ephenodrom committed:
Improve ASN1BitString

mwcw committed:
Merge branch 'tekartikdev-master' into master

mwcw committed:
Merge branch 'master' of https://github.com/Ephenodrom/pc-dart into Ephenodrom-master

mwcw committed:
Merge branch 'Ephenodrom-master' into master

mwcw committed:
Merge branch 'rsa-public-exponent' of https://github.com/hoylen/pc-dart into hoylen-rsa-public-exponent

vHanda committed:
Add missing secp384r1 identifier

I'm using PointyCastle to generate secp384r1 keys.

schancel committed:
Ensure negative positive BigInts are padded if their first byte has 0x80

Encoding a BigInt in 2s-compliment requires an extra padding byte if the
0x80 bit is set in the first byte position. This commit checks to see if
that's the case, and adds the padding to ensure it is parseable.

Ephenodrom committed:
Remove unused dart:io import

mwcw committed:
Merge branch 'hoylen-rsa-public-exponent' into master

mwcw committed:
Merge branch 'master' of https://github.com/vHanda/pc-dart into vHanda-master

mwcw committed:
Merge branch 'vHanda-master' into master

mwcw committed:
Merge branch 'fix-encoding' of https://github.com/schancel/pc-dart into schancel-fix-encoding

mwcw committed:
Added unsigned encoding and tests for encoding.

mwcw committed:
Merge branch 'schancel-fix-encoding' into master

mwcw committed:
Pull requests:
   #58 Add missing secp384r1 identifier
   #54 RSAPrivateKey calculates the public exponent from the other values.
   #59 Ensure negative positive BigInts are padded if their first byte has 0x80

NB: BigInt handling has changed!
The default BigInt handling functions in the util package now treat encoded BigInts as twos compliment numbers, so:

utils.dart:
- decodeBigInt is twos compliment.
- encodeBigInt adds padding to preserve sign where appropriate.
- encodeBigIntAsUnsigned writes the magnitude without any padding.
- decodeBigIntWithSign allows the specification of an arbitrary sign.

Previous uses of decodeBigInt where the expectation is an unsigned integer have
been updated with decodeBigIntWithSign(1, magnitude).

mwcw committed:
Merge branch 'master' of https://github.com/Ephenodrom/pc-dart into Ephenodrom-master

mwcw committed:
Merge branch 'Ephenodrom-master' into master

mwcw committed:
Updated change log and contributors (thanks!)

mwcw committed:
Updated changelog.

izaera committed:
Update my GitHub URL

Just that :-).

Ephenodrom committed:
Fixed small bug in the decodeLength function

mwcw committed:
Merge branch 'Ephenodrom-master' into master

mwcw committed:
Merge branch 'patch-1' of https://github.com/izaera/pc-dart into izaera-patch-1

mwcw committed:
Merge branch 'izaera-patch-1' into master

mwcw committed:
Update for version 2.0.1

AKushWarrior committed:
Merge pull request #5 from bcgit/master

merge updates from upstream

AKushWarrior committed:
minor README update to clarify initialization procedure

AKushWarrior committed:
move to null safety with dependencies and analysis options

AKushWarrior committed:
begin null safety migration

AKushWarrior committed:
continue null-safety migration (internals, stream ciphers, signers), add markers to touched files)

AKushWarrior committed:
null-safety migration (random, padding)

AKushWarrior committed:
null-safety migration (MAC)

AKushWarrior committed:
key generators

AKushWarrior committed:
null safety: key derivators and updates to MACs

AKushWarrior committed:
null safety: ecc base (and reformatted dir)

AKushWarrior committed:
null safety: digests & tests

AKushWarrior committed:
null safety: aes modes

AKushWarrior committed:
run dartfmt

AKushWarrior committed:
fix lints, strip unnecessary null checks

AKushWarrior committed:
begin work on fixing failing tests

AKushWarrior committed:
continued work on failing tests

AKushWarrior committed:
fix SIC tests

AKushWarrior committed:
fix padded block cipher impl

AKushWarrior committed:
remove markers of null migration

AKushWarrior committed:
cleanup + fix pre-init procedure for cbcmac

AKushWarrior committed:
fix CMAC issue

AKushWarrior committed:
fix RSA keygen issue

AKushWarrior committed:
add info to README, add null-safety to CHANGELOG and pubspec.yaml, tick version to 3.0.0 as per dart guidelines

AKushWarrior committed:
reformat

AKushWarrior committed:
Add tests to all_tests_web, fix bug in HKDF

mwcw committed:
Merge branch 'AKushWarrior-master' into 3.0.0-nullsafety.0

AKushWarrior committed:
Add IGE + tests

AKushWarrior committed:
Update README for IGE

mwcw committed:
Fix warnings from pub tool.

mwcw committed:
Fix warnings from pub tool.

AKushWarrior committed:
starting work on CCM (broken)

AKushWarrior committed:
fix CCM issues

AKushWarrior committed:
logistical problems: README and CHANGELOG

AKushWarrior committed:
add another IGE test from OpenSSL

mwcw committed:
Merge branch 'AKushWarrior-master' into nullsafety

mwcw committed:
Updated pubspec and changelog

Tougee committed:
EdDSA part to null safety
